example/bucket/main.tf

```diff
@@ -34,3 +34,9 @@ resource "couchbase_bucket" "bucket" {
   quota = 150
   type = 0
 }
+
+# Create a primary index
+resource "couchbase_index" "index" {
+  bucket_name = couchbase_bucket.bucket.name
+  index_name = "def_${couchbase_bucket.bucket.name}_primary"
+}
```

Comment:
In this case u should add depends_on = [couchbase_bucket.bucket]. As @astamur already mentioned

Reply:
Not necessary, Terraform allows making transparent dependencies by using attributes of resources if you use the attribute of resource A in the definition of resource B it automatically makes resource B dependent on A

Reply:
Oh, got it)